automerge-wasm/README.md

```diff
@@ -1,696 +1,478 @@
+
+```javascript
+    let doc = create()
+
+    doc.put("_root", "key", "val1")
+
+    doc.value("_root", "key")        // returns ["str","val1"]
+    doc.pendingOps()                 // returns 1
+
+    doc.rollback()
+
+    doc.value("_root", "key")        // returns null
+    doc.pendingOps()                 // returns 0
+
+    doc.put("_root", "key", "val2")
+
+    doc.pendingOps()                 // returns 1
+
+    doc.commit("test commit 1")
+
+    doc.value("_root", "key")        // returns ["str","val2"]
+    doc.pendingOps()                 // returns 0
+
+    doc.free()
+```
+
+### Viewing Old Versions of the Document
+
+All query functions can take an optional argument of `heads` which allow you to query a prior document state. Heads are a set of change hashes that uniquely identify a point in the document history.  The `getHeads()` method can retrieve these at any point.
+
+```javascript
+    let doc = create()
+
+    doc.put("_root", "key", "val1")
+    let heads1 = doc.getHeads()
+
+    doc.put("_root", "key", "val2")
+    let heads2 = doc.getHeads()
+
+    doc.put("_root", "key", "val3")
+
+    doc.value("_root","key")          // returns ["str","val3"]
+    doc.value("_root","key",heads2)   // returns ["str","val2"]
+    doc.value("_root","key",heads1)   // returns ["str","val1"]
+    doc.value("_root","key",[])       // returns null
+
+    doc.free()
+```
+
+This works for `value()`, `values()`, `keys()`, `length()`, `text()`, and `materialize()`
+
+Queries of old document states are not indexed internally and will be slower than normal access.  If you need a fast indexed version of a document at a previous point in time you can create one with `doc.forkAt(heads, actor?)`
+
+### Forking and Merging
+
+You can `fork()` a document which makes an exact copy of it.  This assigns a new actor so changes made to the fork can be merged back in with the original.  The `forkAt()` takes a Heads, allowing you to fork off a document from a previous point in its history.  These documents allocate new memory in WASM and need to be freed.
+
+The `merge()` command applies all changes in the argument doc into the calling doc.  Therefore if doc a has 1000 changes that doc b lacks and doc b has only 10 changes that doc a lacks, `a.merge(b)` will be much faster than `b.merge(a)`.
+
+```javascript
+    let doc1 = create()
+    doc1.put("_root", "key1", "val1")
+
+    let doc2 = doc1.fork()
+
+    doc1.put("_root", "key2", "val2")
+    doc2.put("_root", "key3", "val3")
+
+    doc1.merge(doc2)
+
+    doc1.materialize("_root")       // returns { key1: "val1", key2: "val2", key3: "val3" }
+    doc2.materialize("_root")       // returns { key1: "val1", key3: "val3" }
+
+    doc1.free(); doc2.free()
+```
+
+Note that calling `a.merge(a)` will produce an unrecoverable error from the wasm-bindgen layer which (as of this writing) there is no workaround for.
+
+### Saving and Loading
```

Comment:
We should document somewhere in this section that the concatenation of two automerge documents is the merge of those documents because it's just too neat to not shout about it.